src/SignalR/clients/ts/FunctionalTests/ts/Common.ts

```diff
@@ -100,3 +101,11 @@ export function eachTransportAndProtocol(action: (transport: HttpTransportType,
 export function getGlobalObject(): any {
     return typeof window !== "undefined" ? window : global;
 }
+
+export function eachHttpClient(action: (transport: HttpClient) => void) {
+    const httpClients: HttpClient[] = [new FetchHttpClient(TestLogger.instance), new XhrHttpClient(TestLogger.instance), new DefaultHttpClient(TestLogger.instance)];
```

Comment:
This will run fetch twice in environments that support fetch and fail in environments that don't support fetch.

Make this conditionally add the clients if they're supported. Don't forget about the `NodeHttpClient` too, we have an `isNode` function at https://github.com/aspnet/AspNetCore/blob/46db3677602c5527346f7c067fba3b5c742120bb/src/SignalR/clients/ts/signalr/src/Utils.ts#L37